From 8c6abd36764a5633dafc6acece0e6a5beb4e5e11 Mon Sep 17 00:00:00 2001
From: Ahmad Samir <a.samirh78@gmail.com>
Date: Mon, 6 Oct 2025 17:54:20 +0300
Subject: [PATCH 318/553] QLogging: fix narrowing conversion assert

Amends a213dca680827336cb63de743be3f019c4521808. Pointed out by Eddy in
code review.

Pick-to: 6.8 6.5
Change-Id: I061f9892384fe62650dfec23842686d1ba53130a
Reviewed-by: Thiago Macieira <thiago.macieira@intel.com>
(cherry picked from commit a9ced75ee065c60442a9da81487fd9eb8dcdc6af)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/corelib/global/qlogging.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/corelib/global/qlogging.cpp b/src/corelib/global/qlogging.cpp
index 1f29fb48a70..c6d2e262bc0 100644
--- a/src/corelib/global/qlogging.cpp
+++ b/src/corelib/global/qlogging.cpp
@@ -1445,7 +1445,7 @@ Q_NEVER_INLINE void QInternalMessageLogContext::populateBacktrace(int frameCount
 {
     assert(frameCount >= 0);
     BacktraceStorage &result = backtrace.emplace(TypicalBacktraceFrameCount + frameCount);
-    Q_ASSERT(frameCount == int(frameCount));
+    Q_ASSERT(result.size() == int(result.size()));
     int n = ::backtrace(result.data(), int(result.size()));
     if (n <= 0)
         result.clear();
-- 
2.51.2

